ktor-client/ktor-client-core/js/src/io/ktor/client/engine/js/compatible/Utils.kt

```diff
@@ -26,19 +26,38 @@ internal fun readBody(
     response: Response,
     callContext: CoroutineContext
 ): ByteReadChannel = if (PlatformUtils.IS_NODE) {
-    callContext.readBodyBlocking(response)
+    callContext.readBodyNode(response)
 } else {
-    callContext.readBodyStream(response)
+    callContext.readBodyBrowser(response)
 }
 
-private fun CoroutineContext.readBodyBlocking(response: Response): ByteReadChannel = GlobalScope.writer(this) {
-    val responseBuffer = response.arrayBuffer().await()
-    channel.writeFully(Uint8Array(responseBuffer).asByteArray())
+private fun CoroutineContext.readBodyNode(response: Response): ByteReadChannel = GlobalScope.writer(this) {
+    val body = response.body ?: error("Fail to get body")
+    val responseData = Channel<ByteArray>(Channel.UNLIMITED)
+
+    body.on("data") { chunk: ArrayBuffer ->
+        responseData.offer(Uint8Array(chunk).asByteArray())
+    }
+
+    body.on("end") {
+        responseData.close()
+    }
```

Comment:
Shouldn't we listen for [error events](https://nodejs.org/api/stream.html#stream_event_error_1) as well?